pyccel/ast/core.py

```diff
@@ -182,7 +184,7 @@ def broadcast(shape_1, shape_2):
     b = len(shape_2)
     if a>b:
         new_shape_2 = (1,)*(a-b) + tuple(shape_2)
-        new_shape_1 = shape_2
+        new_shape_1 = shape_1
```

Comment:
Was this a bug? How did you catch it? Do we have a test for this now?

Reply:
yes it's a bug , the broadcasting is never tested 
because we don't allow for the moment  something like  `y = x + 1` where `x` is an ndarray
maybe we can think of a case where we need to check the shape

Reply:
I think we discussed the broadcast function in a different merge. My understanding was that there would be a pull request at some point to make it cover more cases and to test it properly. But I don't see an issue for this so maybe I misremembered.
I caught this bug with a very small test I wrote while checking that a different error was raised correctly. Fixing this bug made the one below apparent thanks to our current tests

Reply:
Do we have a unit test that would fail if we change this code back to what it was?
I think that's important.

Reply:
I have just added one